cmd/account/mgmt/account-unassign.go

```diff
@@ -0,0 +1,370 @@
+		destinationOU   string
+		rootID          string
+	)
+
+	// Instantiate Aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+
+	o.awsClient = awsClient
+
+	if o.accountID != "" {
+
+		accountIdList = append(accountIdList, o.accountID)
+
+	}
+
+	if o.username != "" {
+		// Check that username is not a hive
+		if strings.HasPrefix(o.username, "hive") {
+			return fmt.Errorf("non-ccs account provided, only developers account accepted")
+		}
+
+		accountUsername = o.username
+		accountIdList, err = o.listAccountsFromUser(accountUsername)
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Printf("Are you sure you want to unassign the accounts? [y/n] ")
+	reader := bufio.NewReader(os.Stdin)
+
+	response, err := reader.ReadString('\n')
+	if err != nil {
+		return err
+	}
+
+	if response == "y" || response == "Y" {
+		// Delete login profile
+		err := o.deleteLoginProfile(accountUsername)
+		if err != nil {
+			return err
+		}
+		// Delete access keys
+		err = o.deleteAccessKeys(accountUsername)
+		if err != nil {
+			return err
+		}
+		// Delete signing certificates
+		err = o.deleteSigningCert(accountUsername)
+		if err != nil {
+			return err
+		}
+		// Delete policies
+		err = o.deletePolicies(accountUsername)
+		if err != nil {
+			return err
+		}
+		// Delete attached policies
+		err = o.deleteAttachedPolicies(accountUsername)
+		if err != nil {
+			return err
+		}
+		// Delete groups
+		err = o.deleteGroups(accountUsername)
+		if err != nil {
+			return err
+		}
+		// Delete user
+		err = o.deleteUser(accountUsername)
+		if err != nil {
+			return err
+		}
+		// loop through accounts list and untag and move them back into root OU
+		for _, id := range accountIdList {
+
+			err = o.untagAccount(id)
+			if err != nil {
+				return err
+			}
+
+			destinationOU = "r-rs3h-ry0hn2l9"
+			rootID = "r-rs3h"
+			if o.payerAccount == "osd-staging-1" {
+				destinationOU = "ou-0wd6-z6tzkjek"
+				rootID = "ou-0wd6"
+			}
+
+			err = o.moveAccount(id, rootID, destinationOU)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+var ErrNoAccountsForUser error = fmt.Errorf("user has no aws accounts")
+
+func (o *accountUnassignOptions) listAccountsFromUser(user string) ([]string, error) {
+
+	inputFilterTag := &resourcegroupstaggingapi.GetResourcesInput{
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key: aws.String("owner"),
+				Values: []*string{
+					aws.String(user),
+				},
+			},
+		},
+	}
+	accounts, err := o.awsClient.GetResources(inputFilterTag)
+	if err != nil {
+		return []string{}, err
+	}
+
+	if len(accounts.ResourceTagMappingList) == 0 {
+		return []string{}, ErrNoAccountsForUser
+	}
+
+	var accountIdList []string
+	// Get last 12 digits of ResourceARN and append it to account list
+	for _, a := range accounts.ResourceTagMappingList {
+		accountIdList = append(accountIdList, (*a.ResourceARN)[len(*a.ResourceARN)-12:])
+	}
+	if err != nil {
```

Comment:
You forgot to test this situation. Please add this as part of your unit-tests ;)
Not to mention this is condition is always 'false' because 'err' is always 'nil' in this part of the code! Ouch.